### PR TITLE
feat(ingest): add classification to bigquery, redshift

### PR DESCRIPTION
**Pull request by mayurinehate**
Stacked on top of https://github.com/datahub-project/datahub/pull/10013

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by hsheth2 — APPROVED** [review]
Looks reasonable to me

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py [hunk 1]
```diff
@@ -732,12 +750,27 @@ def _process_schema(
...
                 )
+                yield from classification_workunit_processor(
+                    table_wu_generator,
+                    self.classification_handler,
+                    self.data_reader,
+                    [project_id, dataset_name, table.name],
+                    data_reader_kwargs=dict(
+                        sample_size_percent=(
+                            self.config.classification.sample_size
+                            * 1.2
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py:767]: can we extract out this 1.2 constant in a follow up PR?

File: metadata-ingestion/src/datahub/ingestion/source/sql/teradata.py [hunk 2]
```diff
@@ -447,6 +447,11 @@ class TeradataConfig(BaseTeradataConfig, BaseTimeWindowConfig):
 @capability(SourceCapability.LINEAGE_COARSE, "Optionally enabled via configuration")
 @capability(SourceCapability.LINEAGE_FINE, "Optionally enabled via configuration")
 @capability(SourceCapability.USAGE_STATS, "Optionally enabled via configuration")
+@capability(
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/sql/teradata.py:450]: the capability annotations support inheritance, so I don't think these are actually necessary

**Review by hsheth2 — APPROVED** [review]
Two non-blocking comments that we can do in a follow-up

File: metadata-ingestion/src/datahub/ingestion/source/sql/teradata.py (hunk 2 above)
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/sql/teradata.py:450]: wondered why we had no annotation on SQLAlchemySource source. Some like PLATFORM_INSTANCE, DOMAINS, SCHEMA_METADATA, CONTAINERS, DELETION_DETECTION can probably move in there.

**Pull request merged**